integration-test/src/test/java/org/sagebionetworks/IT100TableControllerTest.java

```diff
@@ -305,6 +305,35 @@ public void testCreateTableEntity() throws Exception {
 		
 	}
 
+	@Test
+	public void testCurrentUserFunction() throws Exception{
+		// Create a few columns to add to a table entity
+		ColumnModel one = new ColumnModel();
+		one.setName("userId");
+		one.setColumnType(ColumnType.DOUBLE);
+		one = synapse.createColumnModel(one);
+
+		TableEntity table = createTable(Lists.newArrayList(one.getId()));
+
+		List<ColumnModel> columns = synapse.getColumnModelsForTableEntity(table.getId());
+
+		// Append some rows
+		RowSet set = new RowSet();
+		List<Row> rows = Lists.newArrayList(TableModelTestUtils.createRow(null, null, userId.toString()),
+				TableModelTestUtils.createRow(null, null, "2"),
+				TableModelTestUtils.createRow(null, null, "3"),
+				TableModelTestUtils.createRow(null, null, "4"));
+		set.setRows(rows);
+		set.setHeaders(TableModelUtils.getSelectColumns(columns));
+		set.setTableId(table.getId());
+		synapse.appendRowsToTable(set, MAX_APPEND_TIMEOUT, table.getId());
+
+		assertCountResults("select * from " + table.getId() + " where userId = CURRENT_USER()", table.getId(), 1L);
+		assertQueryResults("select * from " + table.getId() + " where userId = CURRENT_USER()", null, null, table.getId(), (queryResults) -> {
+			assertEquals(userId.toString(), queryResults.getRows().get(0).getValues().get(0));

```

Comment:
you might want to add an assertEquals(1, queryResults.getRows().size()) to check the number of rows and additionally you could check also the number of values: assertEquals(1, queryResults.getRows().get(0).getValues().size()) assuming that your query is something on the lines of "select userId from..."